### PR TITLE
Enable compilation of mock driver and test binary (gtapi)

### PR DESCRIPTION
**Pull request by abelardojarab**
Build gtapi and mock driver; define basic test using combination of mock driver and gtapi.

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  message(STATUS "Trying to fetch gtest throught git")
+  find_package(Git REQUIRED)
+  execute_process(COMMAND "${GIT_EXECUTABLE}"
+    "clone"
+    "--branch"
+    "release-1.8.0"
+    "https://github.com/google/googletest.git"
+    "gtest" WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+  set (gtest_root "${CMAKE_BINARY_DIR}/gtest")
+  message(STATUS "gtest locatet at: ${gtest_root}")
```
Comment (r-rojo) [inline tests/CMakeLists.txt]: misspelled `locateet`

File: cmake/modules/Findjsoncpp.cmake [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# - Try to find libjson-c
```
Comment (r-rojo) [inline cmake/modules/Findjsoncpp.cmake]: Copy/paste error?

File: tests/common_test.h [hunk 3]
```diff
@@ -0,0 +1,308 @@
+/*common_test.h */
+
```
Comment (rpSTTD) [inline tests/common_test.h]: this needs to be refactored to work with rpm files.  Please only select the functions that are needed for the tests.

File: tests/fpgaopen_app.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#include <getopt.h>
```
Comment (rpSTTD) [inline tests/fpgaopen_app.cpp]: This file is not needed for the initial checkin.

File: tests/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+add_dependencies(mock mock-sysfs-prepare)
+target_link_libraries(mock dl safestr)
+
+############################################################################
+## Add testing targets #####################################################
+############################################################################
+
+add_test(NAME test_enum_mock_drv
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+  COMMAND gtapi -j ${CMAKE_BINARY_DIR}/tests/configuration.json --gtest_filter=*EnumCommon*enum*)
```
Comment (rpSTTD) [inline tests/CMakeLists.txt]: Need to have the capability to modify the --gtest_filte.

File: tests/function/gtOpenClose.cpp [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+using namespace common_test;
+using namespace std;
+
+class LibopaecOpenFCommonHW : public common_test::BaseFixture,
+                              public ::testing::Test {};
+
+class LibopaecCloseFCommonHW : public common_test::BaseFixture,
+                               public ::testing::Test {};
+
+TEST_F(LibopaecOpenFCommonHW, 03) {
```
Comment (rpSTTD) [inline tests/function/gtOpenClose.cpp]: All tests that have integer only as test name need to be verified with mock driver first.

File: tests/CMakeLists.txt (hunk 5 above)
Comment (r-rojo) [reply tests/CMakeLists.txt]: Calling `make test` should result in all unit tests (that don't require hardware) being called and are used to assess that the build is not broken. Any other tests that are more in depth can always be called manually by invoking either `gtapi` itself or a wrapper script that maps the kind of test to a specific filter. Those can also be encapsulated by a build target.
If more tests need to be called to assess the health of the build, they should be added to either:
* another call to add_test
* an longer filter

It may still be a good idea to set the filter in a CMake variable though.

File: tests/COPYING [hunk 7]
```diff
@@ -0,0 +1,37 @@
+If not otherwise noted in the individual file header, the OPAE test source code,
+located in the directories
+
+  * unit,
+  * function,
+  * stress,
+  * doc,
+
+is distributed under the following license:
+
```
Comment (luebbers) [inline tests/COPYING]: Aren't these covered with the original COPYING file in the top level directory?

File: cmake/modules/Findjson-c.cmake [hunk 8]
```diff
@@ -27,6 +27,7 @@ find_library(libjson-c_LIBRARIES
   /usr/local/lib
   /usr/lib
   /lib
+  /usr/lib/x86_64-linux-gnu
```
Comment (luebbers) [inline cmake/modules/Findjson-c.cmake]: Is this related to tests?

**Comment by abelardojarab** [comment]
Still over-bloated; next version will be trimmed down.
Apologies.
